### PR TITLE
Fix for #762 - thread pool leak

### LINKED ISSUE #762: Thread pool leak
### Describe the bug
`_messageQueue.PeekAndWait()` (line 352, ManagedMqttClient.cs) causes thread leak due to lock on `ManualResetEvent` when a connection is already dead. Multiple reconnections can cause a thread pool exhaustion (especially on Xamarin.Android)

### Which project is your bug related to?
- ManagedClient

### To Reproduce
1. Using this version of MQTTnet '3.0.8'.
2. Start/stop the client repeatedly
3. See the thread stacks full of `BlockingQueue.PeekAndWait()`

### Expected behavior
Valid cancellation of `PeekAndWait`

### Screenshots
image

### Additional context / logging
This can be easily fixed by using a `ManualResetEventSlim` which supports a `CancellationToken`

### PR DESCRIPTION
**Pull request by zawodskoj**
Fixes #762

### CODE REVIEW

**Comment by zawodskoj** [comment]
There are strange failed test. Should we leave it as is?

**Comment by SeppPenner** [comment]
> There is strange failed test. Should we leave it as is?

Which one is it? I haven't run the build process yet.

**Comment by zawodskoj** [comment]
> Which one is it? I haven't run the build process yet.

```
 X Create_Managed_Client_With_Logger [862ms]
  Error Message:
   Assert.IsFalse failed. 
  Stack Trace:
     at MQTTnet.Tests.MqttFactory_Tests.Create_Managed_Client_With_Logger() in C:\projects\mqttnet\Tests\MQTTnet.Core.Tests\MqttFactory_Tests.cs:line 73
```

**Comment by SeppPenner** [comment]
Ok, I will check this later.

File: Source/MQTTnet/Internal/BlockingQueue.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public class BlockingQueue<TItem>
     {
         private readonly object _syncRoot = new object();
         private readonly LinkedList<TItem> _items = new LinkedList<TItem>();
-        private readonly ManualResetEvent _gate = new ManualResetEvent(false);
+        private readonly ManualResetEventSlim _gate = new ManualResetEventSlim(false);
```
Comment (chkr1011) [inline Source/MQTTnet/Internal/BlockingQueue.cs]: ManualResetEventSlim is Disposable. So please implement the Dispose interface. Also parent classes might need to call Dispose then as well.

Comment (zawodskoj) [reply Source/MQTTnet/Internal/BlockingQueue.cs]: Ok, i'll do it later

Comment (chkr1011) [reply Source/MQTTnet/Internal/BlockingQueue.cs]: I have some time at the moment. I will do. Hope you did not start already 🙈

**Pull request merged**